postpic/datahandling.py

```diff
@@ -294,8 +294,18 @@ def label(self):
 
     def _value_to_index_nonlinear(self, value):
         if self._inv_map is None:
-            self._inv_map = spinterp.interp1d(self.grid, np.linspace(0, len(self)-1, len(self)),
-                                              bounds_error=False, fill_value='extrapolate')
+            grid_and_nodes = np.zeros(2*len(self)+1)
+
+            grid_and_nodes[::2] = self.grid_node
+            grid_and_nodes[1::2] = self.grid
+
+            indices_grid_and_nodes = np.linspace(-0.5, len(self)-0.5, len(grid_and_nodes))
```

Comment:
Ist es eine gute idee die daten vom Grid UND von den Grid_nodes zu nehmen? Im `__init__` werden die Daten so gelegt, dass der Grid punkt immer genau mittig zwischen zwei grid_nodes ist. Wenn das nun hier interpoliert wird, dann bekommt man immer eine schnelle frequenz auf die interpolation. Ich wuerde vorschlagen ausschliesslich die grid_nodes zu verwenden und nicht das grid selbst.

Reply:
Das verändert dann aber das Verhalten bei nicht-linearen Grids, insbesondere landen die Gitterpunkte nicht mehr genau auf den von "grid" definierten werten. Meine Logik war daher, dass ich versuche sowohl grid, als auch grid_node exakt zu treffen, um in allen Fällen vorhersagbare Ergebnisse zu erzeugen.

Im Optimalfall müsste man vielleicht beim erzeugen der Axis die Information hinterlegen, welches aus {grid, grid_node} explizit gegeben wurde(n) und diese für die Interpolation nehmen.

Alternativ könnte man bereits beim erzeugen der Axis für das nicht gegebene aus {grid, grid_node} bereits spline-interpolation nehmen statt genau mittiger punkte (sollte ja für lineare gitter auf das selbe raus kommen), dann sollte auch für nicht-lineare gitter alles konsistent bleiben, ohne dass eine solche oberschwingung auf den koordinaten zu erzeugen.

Reply:
Hmm... Mein Gedankengang war eher, dass `grid_nodes` die vollstaendige information tragen, denn sie definieren genau welche Postitionen auf welche Gitterpunkte fallen. Wenn jemand mit den `grid_nodes` arbeitet, bekommt man reproduzierbare Ergebnisse und wenn man aber nur das `grid` verwendet (also unvollstaendige information), muss man sich nicht wundern, dass es nicht reproduzierbar ist. Ganz deutlich wird das wenn es nur einen Gitterpunkt gibt (also `len(grid)==1` aber `len(grid_node)==2`). Insofern waere es dann inkosistent die (redundante) information vom `grid` auch mit einzubeziehen.... In dieser Vorstellung gibt es das `grid` nur, weil es zum plotten praktisch ist.